core/src/main/java/io/kestra/core/server/AbstractServiceLivenessCoordinator.java

```diff
@@ -0,0 +1,131 @@
+package io.kestra.core.server;
+
+import io.kestra.core.repositories.ServiceInstanceRepositoryInterface;
+import io.micronaut.core.annotation.Introspected;
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Random;
+
+import static io.kestra.core.server.Service.ServiceState.CREATED;
+import static io.kestra.core.server.Service.ServiceState.RUNNING;
+
+/**
+ * Base class for coordinating service liveness.
+ */
+@Introspected
+@Slf4j
+public abstract class AbstractServiceLivenessCoordinator extends AbstractServiceLivenessTask {
+
+    private final static int DEFAULT_SCHEDULE_JITTER_MAX_MS = 500;
+
+    private static final String TASK_NAME = "service-liveness-coordinator-task";
+
+    protected final ServiceInstanceRepositoryInterface serviceInstanceRepository;
+
+    // mutable for testing purpose
+    protected String serverId = ServerInstance.INSTANCE_ID;
+
+    /**
+     * Creates a new {@link AbstractServiceLivenessCoordinator} instance.
+     *
+     * @param serviceInstanceRepository The {@link ServiceInstanceRepositoryInterface}.
+     * @param serverConfig              The server configuration.
+     */
+    @Inject
+    public AbstractServiceLivenessCoordinator(final ServiceInstanceRepositoryInterface serviceInstanceRepository,
+                                              final ServerConfig serverConfig) {
+        super(TASK_NAME, serverConfig);
+        this.serviceInstanceRepository = serviceInstanceRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     **/
+    @Override
+    protected Duration getScheduleInterval() {
+        // Multiple Executors can be running in parallel. We add a jitter to
+        // help distributing the load more evenly among the ServiceLivenessCoordinator.
+        // This is also used to prevent all ServiceLivenessCoordinator from attempting to query the repository simultaneously.
+        Random r = new Random(); //NOSONAR
```

Comment:
Please use a RandomGenerator which is more performant than a Random.
I also think it can be shared across threads.